sentinel-core/src/main/java/com/alibaba/csp/sentinel/slots/block/degrade/DegradeSlot.java

```diff
@@ -15,30 +15,73 @@
  */
 package com.alibaba.csp.sentinel.slots.block.degrade;
 
+import java.util.List;
+
+import com.alibaba.csp.sentinel.Entry;
 import com.alibaba.csp.sentinel.context.Context;
 import com.alibaba.csp.sentinel.node.DefaultNode;
 import com.alibaba.csp.sentinel.slotchain.AbstractLinkedProcessorSlot;
 import com.alibaba.csp.sentinel.slotchain.ProcessorSlot;
 import com.alibaba.csp.sentinel.slotchain.ResourceWrapper;
+import com.alibaba.csp.sentinel.slots.block.BlockException;
+import com.alibaba.csp.sentinel.slots.block.degrade.circuitbreaker.CircuitBreaker;
 import com.alibaba.csp.sentinel.spi.SpiOrder;
+import com.alibaba.csp.sentinel.util.TimeUtil;
 
 /**
- * A {@link ProcessorSlot} dedicates to {@link DegradeRule} checking.
+ * A {@link ProcessorSlot} dedicates to circuit breaking.
  *
- * @author leyou
+ * @author Carpenter Lee
+ * @author Eric Zhao
  */
 @SpiOrder(-1000)
 public class DegradeSlot extends AbstractLinkedProcessorSlot<DefaultNode> {
 
     @Override
-    public void entry(Context context, ResourceWrapper resourceWrapper, DefaultNode node, int count, boolean prioritized, Object... args)
-        throws Throwable {
-        DegradeRuleManager.checkDegrade(resourceWrapper, context, node, count);
+    public void entry(Context context, ResourceWrapper resourceWrapper, DefaultNode node, int count,
+                      boolean prioritized, Object... args) throws Throwable {
+        performChecking(resourceWrapper);
+
         fireEntry(context, resourceWrapper, node, count, prioritized, args);
     }
 
+    void performChecking(ResourceWrapper r) throws BlockException {
+        List<CircuitBreaker> circuitBreakers = DegradeRuleManager.getCircuitBreakers(r.getName());
+        if (circuitBreakers == null || circuitBreakers.isEmpty()) {
+            return;
+        }
+        for (CircuitBreaker cb : circuitBreakers) {
+            if (!cb.tryPass()) {
+                throw new DegradeException(cb.getRule().getLimitApp(), cb.getRule());
+            }
+        }
+    }
+
     @Override
-    public void exit(Context context, ResourceWrapper resourceWrapper, int count, Object... args) {
-        fireExit(context, resourceWrapper, count, args);
+    public void exit(Context context, ResourceWrapper r, int count, Object... args) {
```

Comment:
And here's a problem that if something went wrong which may cause RT down to more than 10s (100ms for normal) . Our breaker will cut the traffic after 10s. Maybe we could improve it (or as a new issue and do it in future) that unfinished entries could be recorded as `statIntervalMs` when slidingCounter is reset.

Reply:
Yes, it's necessary to improve the scenarios regarding slow in-flight requests. See #1405

Reply:
Oh they are not the same issue i think.

How to calculate a more reasonable RT is one thing while how to make a more reasonable degradation is quite another thing. The primary reason is that the calculating window is separated from general RT, isn't?

And sure we can improve it later but it's more important than making it more reasonable calculating RT. Maybe another time window is required to make it under tracing.

Reply:
We could open a new issue to discuss it.